presto-jdbc/src/main/java/com/facebook/presto/jdbc/QueryInterceptor.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.jdbc;
+
+import java.sql.Statement;
+import java.util.Map;
+import java.util.Optional;
+
+public interface QueryInterceptor
+{
+    /**
+     * Called once per {@link <PrestoConnection>} at instantiation.
+     *
+     * @param properties the session properties supplied in the URI
+     */
+    void init(Map<String, String> properties);
```

Comment:
thinking a bit more - how do you feel about having a default empty implementation for this as well. The caller can choose to implement it or ignore the - the interface itself does not require an implementation to implement this function.

Reply:
Good point, I like it